netadapter/server/grpcserver/grpc_server.go

```diff
@@ -0,0 +1,130 @@
+package grpcserver
+
+import (
+	"context"
+	"fmt"
+	"net"
+
+	"google.golang.org/grpc/peer"
+
+	"github.com/kaspanet/kaspad/netadapter/server"
+	"github.com/kaspanet/kaspad/netadapter/server/grpcserver/protowire"
+	"github.com/kaspanet/kaspad/util/panics"
+	"github.com/pkg/errors"
+	"google.golang.org/grpc"
+)
+
+type gRPCServer struct {
+	onConnectedHandler server.OnConnectedHandler
+	connections        map[string]*gRPCConnection
+	listeningAddrs     []string
+	server             *grpc.Server
+}
+
+// NewGRPCServer creates and starts a gRPC server with the given
+// listening port
+func NewGRPCServer(listeningAddrs []string) (server.Server, error) {
+	s := &gRPCServer{
+		server:         grpc.NewServer(),
+		listeningAddrs: listeningAddrs,
+		connections:    map[string]*gRPCConnection{},
+	}
+	protowire.RegisterP2PServer(s.server, newP2PServer(s))
+
+	return s, nil
+}
+
+func (s *gRPCServer) Start() error {
+	for _, listenAddr := range s.listeningAddrs {
```

Comment:
Please extract the contents of this loop to a function called listen() or something similar.